server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -321,8 +327,15 @@ private static NumericType resolveNumericType(String value) {
 
     @Override
     public SortFieldAndFormat build(QueryShardContext context) throws IOException {
+        final boolean reverse = order == SortOrder.DESC;
+
         if (DOC_FIELD_NAME.equals(fieldName)) {
-            return order == SortOrder.DESC ? SORT_DOC_REVERSE : SORT_DOC;
+            return reverse ? SORT_DOC_REVERSE : SORT_DOC;
+        } else if (SHARD_DOC_FIELD_NAME.equals(fieldName)) {
+            if (context.getShardRequestIndex() == -1) {
+                throw new IllegalArgumentException("");
```

Comment:
should we put any message in this exception?

Reply:
It's a leftover, the shard request index is guaranteed to be greater than 0 at this point. I removed the check and the exception in https://github.com/elastic/elasticsearch/pull/66093/commits/5af84f589bb04c8b63798f5d8c2ce1522af3b912